tools/src/main/java/org/apache/kafka/tools/BrokerApiVersionsCommand.java

```diff
@@ -0,0 +1,337 @@
+                        client.poll(time.timer(Long.MAX_VALUE));
+                    }
+                } catch (Throwable t) {
+                    LOGGER.error("admin-client-network-thread exited", t);
+                } finally {
+                    pendingFutures.forEach(future -> {
+                        try {
+                            future.raise(Errors.UNKNOWN_SERVER_ERROR);
+                        } catch (IllegalStateException ignored) {
+                        }
+                    });
+                    pendingFutures.clear();
+                }
+            }, "admin-client-network-thread");
+            networkThread.setDaemon(true);
+            networkThread.start();
+        }
+
+        private AbstractResponse send(Node target, AbstractRequest.Builder<?> request) throws InterruptedException {
+            RequestFuture<ClientResponse> future = client.send(target, request);
+            pendingFutures.add(future);
+            future.awaitDone(Long.MAX_VALUE, TimeUnit.MILLISECONDS);
+            pendingFutures.remove(future);
+            if (future.succeeded()) {
+                return future.value().responseBody();
+            } else {
+                throw future.exception();
+            }
+        }
+
+        private AbstractResponse sendAnyNode(AbstractRequest.Builder<?> request) {
+            for (Node broker : bootstrapBrokers) {
+                try {
+                    return send(broker, request);
+                } catch (AuthenticationException e) {
+                    throw e;
+                } catch (Exception e) {
+                    LOGGER.debug("Request {} failed against node {}", request.apiKey(), broker, e);
+                }
+            }
+            throw new RuntimeException("Request " + request.apiKey() + " failed on brokers " + bootstrapBrokers);
+        }
+
+        private KafkaFuture<NodeApiVersions> getNodeApiVersions(Node node) {
+            final KafkaFutureImpl<NodeApiVersions> future = new KafkaFutureImpl<>();
+            try {
+                ApiVersionsResponse response = (ApiVersionsResponse) send(node, new ApiVersionsRequest.Builder());
+                Errors error = Errors.forCode(response.data().errorCode());
+                if (error.exception() != null) {
+                    future.completeExceptionally(error.exception());
+                } else {
+                    future.complete(new NodeApiVersions(response.data().apiKeys(), response.data().supportedFeatures(), response.data().zkMigrationReady()));
+                }
+            } catch (Exception e) {
+                future.completeExceptionally(e);
+            }
+
+            return future;
+        }
+
+        public void awaitBrokers() {
+            List<Node> nodes;
+            do {
+                nodes = findAllBrokers();
+                if (nodes.isEmpty()) {
+                    try {
+                        Thread.sleep(50);
+                    } catch (InterruptedException e) {
```

Comment:
not sure why we need to catch it here. could you please throw it directly as it is a command-line tool